examples/custom.py

```diff
@@ -1,11 +1,16 @@
+from typing import TYPE_CHECKING
+
 from astroid import nodes
 
 from pylint.checkers import BaseChecker
 from pylint.interfaces import IAstroidChecker
 
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
```

Comment:
Well it could result in circular dependency later. If something is only used for typing I think using a typing guard is the way to go (especially with PyLinter that is used "everywhere" and use "everything")

Reply:
I don't really think that's going to be an issue here. The checkers itself are already quite isolated by design. IMO guarding the import is unnecessary work. _We can always change it later if it should become necessary. Although I don't believe it will._

Reply:
> guarding the import is unnecessary work.

The work is already done and reverting = more work.